### PR TITLE
Adding index.hh detection

### PR DESCRIPTION
**Pull request by cardmagic**

### CODE REVIEW

**Comment by henrikbjorn** [comment]
I do not think it is needed to include a default config file for index.hh, the reason being most web projects will have a dedicated webroot directory in order to not expose library code.

Otherwise this is  :+1: from me :)

File: bin/compile [hunk 1]
```diff
@@ -10,6 +10,14 @@ COMPOSER_URL="http://getcomposer.org/composer.phar"
 source $BASE_DIR/common
 source $BASE_DIR/linter/lint
 
+if [ -f "$BUILD_DIR/composer.json" ] || [ -f "$BUILD_DIR/index.php" ]; then
+  TYPE=php
+fi
+
+if [ -f "$BUILD_DIR/index.hh" ]; then
```
Comment (ClaudiuC) [inline bin/compile:17]: Actually, the extension can be anything. By default we support `.php` (note that .hh is actually a C++ header extension). 

The actual way to detect this is to check for the opening tag to be `<?hh`

Feel free to follow up on https://www.facebook.com/groups/hhvm.dev/226724094183364/?notif_t=group_comment_reply for details :)

**Comment by ClaudiuC** [comment]
Thanks for the PR @cardmagic! Please see the comment inline, other than that this looks good to me.

Edit: had a comment about the CLA but we won't need one for this repo :)

**Comment by ptarjan** [comment]
@ClaudiuC actually we support .php and .hh in the type checker and runtime. If you name it anything else we ignore it.

**Comment by henrikbjorn** [comment]
Now that heroku have their own buildpack i am wondering if this have become obsolete?

**Pull request merged**